Mathlib/CategoryTheory/Monoidal/Cartesian/Mon_.lean

```diff
@@ -81,7 +81,7 @@ def Mon_Class.ofRepresentableBy (F : Cᵒᵖ ⥤ MonCat.{w}) (α : (F ⋙ forget
 alias Mon_ClassOfRepresentableBy := Mon_Class.ofRepresentableBy
 
 /-- If `M` is a monoid object, then `Hom(X, M)` has a monoid structure. -/
-abbrev Hom.monoid : Monoid (X ⟶ M) where
+instance Hom.instMonoid : Monoid (X ⟶ M) where
```

Comment:
Do we really want this to be a global instance? Maybe scoped in `Mon_Class`?

Reply:
What could it conflict with?

Reply:
`Monoid (X ⟶ X)` as the monoid of endomorphisms, `Monoid (M ⟶ N)` in `CommGrp` etc.

Reply:
For the first one, we have `CategoryTheory.End`. For the second one, they are defeq, right?

I don't see why the rules in category theory land should be different from outside

Reply:
They are equal but I am not sure if they are defeq. More generally there are preadditive categories where they are additive group structures (which will conflict once we additivize things) on hom sets where you have no control over the defeqs.

Reply:
Since you already open `Mon_Class` everywhere, I don't think scoping it there would be that bad?

Reply:
Hmm actually how do I scope an *instance*? I tried
```
scoped[Mon_Class] attribute [local instance] Hom.instMonoid
-- elaboration function for 'Mathlib.Tactic.scopedNS' has not been implemented
```

Reply:
I agree that the (add)monoid/group instances must be scoped. If you make `instMonoid` a def instead, it seems the syntax would be:
```lean
scoped [Hom_Class] attribute [instance] Hom.instMonoid
```